src/python/pants/backend/go/util_rules/goroot.py

```diff
@@ -119,6 +130,7 @@ async def setup_goroot(golang_subsystem: GolangSubsystem, go_bootstrap: GoBootst
                     level=LogLevel.DEBUG,
                     cache_scope=ProcessCacheScope.PER_RESTART_SUCCESSFUL,
                     env={"GOPATH": "/does/not/matter"},
+                    docker_image=env_tgt.docker_image,
```

Comment:
I don't think that this is what we'll want to do here. As mentioned in https://github.com/pantsbuild/pants/pull/16758#issuecomment-1235808605, I think that we're going to want to adjust the `Process` intrinsic to essentially request something like `ProcessEnvironmentDefaults` (strawname, but: the computed-from-the-environment default values of fields like `platform` and `docker_image`), and inherently apply those defaults.

So 99% of the edits to add a `docker_image` argument in this PR should be removed, I think, as 99% of the time this should be defaulted.

I expect that the only time when someone might explicitly set a `docker_image` might be in the docker backend/plugin, in order to take advantage of the native support in some way (I'm not sure how... @kaos would have a better idea of what might be useful there).

Reply:
Ah the timing. Would've loved to be more involved in getting native support for Docker in Pants. I'm slowly getting into our new life here so are beginning to find some time alas not a lot :)